src/main/java/graphql/language/AstPrinter.java

```diff
@@ -712,7 +712,16 @@ private static void printImpl(StringBuilder writer, Node node, boolean compactMo
      *
      * @param <T> the type of node
      */
-    private interface NodePrinter<T extends Node> {
+    protected interface NodePrinter<T extends Node> {
```

Comment:
can mark this either either as Internal or make it package protected instead of protected